### PR TITLE
[KYUUBI #2207] Support newly added spark data types: DayTimeIntervalType/YearMonthIntervalType

### PR DESCRIPTION
**Pull request by Nick-0723**
### _Why are the changes needed?_
#2207
Support newly added datatypes
spark3.2 : DayTimeIntervalType/YearMonthIntervalType 

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala [hunk 1]
```diff
@@ -39,10 +39,13 @@ object SchemaHelper {
     case _: DecimalType => TTypeId.DECIMAL_TYPE
     case DateType => TTypeId.DATE_TYPE
     case TimestampType => TTypeId.TIMESTAMP_TYPE
+    case tntz if tntz.simpleString.equals("timestamp_ntz") => TTypeId.TIMESTAMP_TYPE
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala:42]: Do we need also handle this in `toJavaSQLType`

Comment (yaooqinn) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala:42]: getDecimalDigits/getDecimalDigits etc

**Comment by yaooqinn** [comment]
Hi @Nick-0723 I check PR and it looks OK to me in general.  For TimestampNTZType, I guess there is still something incomplete, so maybe we shall fix it in another PR to minimize the scope of this one? WDYT？

**Comment by Nick-0723** [comment]
> Hi @Nick-0723 I check PR and it looks OK to me in general. For TimestampNTZType, I guess there is still something incomplete, so maybe we shall fix it in another PR to minimize the scope of this one? WDYT？

I agree ，so do we need revert the TimestampNTZType's code for this PR ?

**Comment by yaooqinn** [comment]
Yes, better revert this part for now

**Review by ulysses-you — APPROVED** [review]

**Review by yaooqinn — APPROVED** [review]
LGTM

**Comment by yaooqinn** [comment]
thanks, merged to master

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala (hunk 1 above)
Comment (Nick-0723) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala]: Is TimestampNTZType's hive thrift type TTypeId.TIMESTAMP_TYPE and is jdbc type TIMESTAMP ？

Comment (yaooqinn) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala]: Ideally，you are right. but there are some historical reasons. at the very beginning，hive does not support timestamp with local timezone but spark support，and they both have only one and be treated as same. now，they both have 2 types supported and the problem caused by old mapping stands out. I guess we shall learn how spark deal with this mess.

Comment (Nick-0723) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala]: The implication is that we need to change the previous Timestamp type to TIMESTAMPLOCALTZ_TYPE ？
```scala
case TimestampType => TTypeId.TIMESTAMPLOCALTZ_TYPE
```

Comment (Nick-0723) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala]: > The implication is that we need to change the previous Timestamp type to TIMESTAMPLOCALTZ_TYPE ？
> 
> ```scala
> case TimestampType => TTypeId.TIMESTAMPLOCALTZ_TYPE
> ```

But it seems that our current hive version (2.3.9) does not support it yet

**Pull request closed without merging**